articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,455 @@
+
+#### Arguments
+
+Any arguments to the Q# operation are simply passed as additional arguments afer the target machine.
+Hence the results of `MeasureSuperpositionArray` on `n=4` qubits would fetched via 
+
+```csharp
+            var multi_qubit_result = MeasureSuperpositionArray.Run(sim, 4).Result;
+```
+
+A full C# host program could thus look like
+
+```csharp
+using System;
+using Microsoft.Quantum.Simulation.Simulators;
+using NamespaceName;
+
+namespace host
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            using var sim = new QuantumSimulator();
+
+            var single_qubit_result = MeasureSuperposition.Run(sim).Result;
+            var multi_qubit_result = MeasureSuperpositionArray.Run(sim, 4).Result;
+
+            Console.WriteLine($"Single qubit result: {single_qubit_result}");
+            Console.WriteLine($"Multiple qubits result: {multi_qubit_result}");
+        }
+    }
+}
+```
+
+At the location of the C# file, the host program can be run from the command line by entering
+
+```bash
+dotnet run
+```
+
+and in this case you will see output written to the console similar to 
+
+```bash
+Single qubit result: Zero
+Multiple qubits result: [One,One,Zero,Zero]
+```
+
+> [!NOTE]
+> Due to the compiler's interoperability with namespaces, we could alternatively make our Q# operations available without the `using NamespaceName;` statement, and simply matching the C# namespace title to it.
+> That is, replacing `namespace host` with `namespace NamespaceName`.
+
+***
+
+## Q# Jupyter Notebooks
+Q# Jupyter Notebooks make use of the IQ# kernel, which allows you to define, compile, and run Q# operations in a single notebook---all alongside instructions, notes, and other content.
+This means that while it is possible to import and use the contents of `*.qs` Q# files, they are not necessary in the execution model.
+
+Here, we will detail how to run the Q# operations defined above, but another good introduction to using Q# Jupyter Notebooks is provided at [Intro to Q# and Jupyter Notebook](https://github.com/microsoft/Quantum/blob/master/samples/getting-started/intro-to-iqsharp/Notebook.ipynb).
+
+### Defining operations
+
+In a Q# Jupyter Notebook, you enter Q# code just as we would from inside the namespace of a Q# file.
+
+So, we can enable access to operations from the [Q# standard libraries](xref:microsoft.quantum.qsharplibintro) with `open` statements for their respective namespaces.
+Upon running a cell with such a statement, the definitions from those namespaces are available throughout the workspace.
+
+> [!NOTE]
+> Operations from [Microsoft.Quantum.Intrinsic](xref:microsoft.quantum.intrinsic) and [Microsoft.Quantum.Canon](xref:microsoft.quantum.canon) (e.g. [`H`](xref:microsoft.quantum.intrinsic.h) and [`ApplyToEach`](xref:microsoft.quantum.canon.applytoeach)) are automatically available in Q# Jupyter Notebooks.
```

Comment:
Would it be good to clarify that this is true within cells, and not for code brought in from external Q# source files?